src/scipp/compat/pandas_compat.py

```diff
@@ -3,53 +3,108 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Union
+from typing import TYPE_CHECKING, Callable, Iterable, Literal, Optional, Tuple, Union
 
-from .._scipp.core import DataArray, Dataset, Variable
+from ..core import DataArray, Dataset, Unit, array
 from ..typing import VariableLike
+from ..units import default_unit
 
 if TYPE_CHECKING:
     import pandas as pd
 
 
-def from_pandas_series(se: pd.Series) -> DataArray:
+def from_pandas_series(
+    se: pd.Series, *, include_index: bool = True, header_parser: HeaderParserArg = None
+) -> DataArray:
     row_index = se.axes[0]
     row_index_name = "row" if row_index.name is None else str(row_index.name)
-    name = "" if se.name is None else str(se.name)
+    name, unit = (
+        ("", default_unit)
+        if se.name is None
+        else _parse_header(str(se.name), header_parser)
+    )
```

Comment:
It is a bit odd to have "defaults" in two places (here and in `_parse_header`). Maybe
```suggestion
    name, unit = _parse_header('' if se.name is None else str(se.name), header_parser)
```
would avoid this?

Reply:
This changed behaviour. Currently, if there is no name, it uses `default_unit` regardless of the parser. With the bracket parser, this change would mean that is uses `unit=None` with the bracket parser.
I think this makes sense.